pkg/scheduler/core/extender.go

```diff
@@ -94,10 +97,47 @@ func NewHTTPExtender(config *schedulerapi.ExtenderConfig) (algorithm.SchedulerEx
 	}, nil
 }
 
+func (h *HTTPExtender) SupportsPreemption() bool {
+	return len(h.preemptVerb) > 0
+}
+
+// ProcessPreemption returns filtered candidate nodes and victims after running preemption logic in extender.
+func (h *HTTPExtender) ProcessPreemption(
+	pod *v1.Pod,
+	nodeToVictims map[string]*schedulerapi.Victims,
```

Comment:
This assumes that extender is NodeCacheCapable (its only given the name of the node). Can we assume the extender needs to support node cache to support preemption? I am ok with making such assumption. We should enforce this check on the configuration.

Reply:
Yep, it seems this assumption is needed if extender want to do preemption.

Reply:
I am not sure if it is the responsibility of scheduler to enforce node cache capability of the extender. For example, what if I want to write a scheduler extender that makes scheduling decision solely based on the name of nodes? I think we should leave it up to the extender to decide how it wants to implement its logic on its own side. So, as long as it exposes the preemption verb, we should assume it supports preemption.

Reply:
My 2 cents is `return len(h.preemptVerb) > 0 && h.nodeCacheCapable` do make sense here, as the extenders' filters are indeed unresolvable if `nodeCacheCapable` is not supported.  Any exception we do not want to rely on this assumption?

Reply:
> the extenders' filters are indeed unresolvable if nodeCacheCapable is not supported

Why is that? What if extender's filters do no need much information from the API server?
 For example, an extender check could be as simple as "node name contains character 'a'"?

Reply:
I am fine with not enforcing that check.

Reply:
> For example, an extender check could be as simple as "node name contains character 'a'"?

I mean this predicate should be unresolvable by preemption, that's why I say `h.nodeCacheCapable` equivalent to nonpreemptable

Anyway I'm totally OK to remove this assumption.

Reply:
While cc @ravigadde , if we do not enforce the check, we will need to pass map[v1.Node]*schedulerapi.Victims` to `ProcessPreempt` instead of only using node names as key. Could you please confirm with this?

Reply:
While need to cc @ravigadde , if we do not enforcing node cache in extender's preemption, we will need to revert to use `v1.Node`  instead of name in ProcessPreempt(), since filter will be checked with given node and pod to ensure preemption of default scheduler works.

Or, we will always allow extender preemption in that case and regardless if the preemptor is allowed by the extender, which is close to https://github.com/kubernetes/kubernetes/pull/58717/files#r164007752

Reply:
IMO, we can keep the logic of sending node names instead of node objects (v1.Node) to extenders. I tried to explain before. We should keep it up to the extenders to decide how they want to get Nodes and Pods information from the API server.